install/certmonger/ipa-server-guard.in

```diff
@@ -35,11 +35,24 @@ import six
 from ipapython import ipautil
 from ipaserver.install import certs
 
+# Return codes. Names of the constants are taken from
+# https://git.fedorahosted.org/cgit/certmonger.git/tree/src/submit-e.h
+OPERATION_NOT_SUPPORTED_BY_HELPER = 6
+
 
 def main():
     if len(sys.argv) < 2:
         raise RuntimeError("Not enough arguments")
 
+    # Avoid the lock if the operation is unsupported by ipa-submit
+    operation = os.environ.get('CERTMONGER_OPERATION')
```

Comment:
Can you add a bit more commentary about why this was a performance issue, i.e. what is Certmonger / the renewal helper doing that causes these operations to be attempted?

Reply:
A certmonger CA can handle a number of different optional operations: 

IDENTIFY
FETCH-ROOTS
GET-SUPPORTED-TEMPLATES
GET-DEFAULT-TEMPLATE
GET-NEW-REQUEST-REQUIREMENTS
GET-RENEW-REQUEST-REQUIREMENTS
FETCH-SCEP-CA-CAPS
FETCH-SCEP-CA-CERTS

ipa-submit only supports a few of these. Rather than obtaining the global lock and forking out to ipa-submit to find out that the operation isn't support, quit early in ipa-server-guard. This will save a lock and a fork.